code/controllers/subsystems/virtual_reality.dm

```diff
@@ -57,34 +57,81 @@
 		ckey = vr_mob.ckey
 		vr_mob.ckey = null
 		vr_mob.old_mob = null
-		vr_mob.languages = list(LANGUAGE_TCB)
+		vr_mob.vr_mob_exit_languages()
 		vr_mob = null
 		to_chat(src, span("notice", "System exited safely, we hope you enjoyed your stay."))
 	if(old_mob)
 		old_mob.ckey = ckey
 		ckey = null
 		old_mob.vr_mob = null
-		languages = list(LANGUAGE_TCB)
+		languages = list(all_languages[LANGUAGE_TCB])
+		to_chat(old_mob, span("notice", "System exited safely, we hope you enjoyed your stay."))
+		old_mob = null
+	if(!vr_mob && !old_mob)
+		to_chat(src, span("danger", "Interface error, you cannot exit the system at this time."))
+		to_chat(src, span("warning", "Ahelp to get back into your body, a bug has occurred."))
+
+/mob/living/silicon/body_return()
+	set name = "Return to Body"
+	set category = "IC"
+
+	if(!vr_mob && !old_mob)
+		return
+
+	if(vr_mob)
+		ckey = vr_mob.ckey
+		vr_mob.ckey = null
+		vr_mob.old_mob = null
+		vr_mob.vr_mob_exit_languages()
+		vr_mob.languages = list(all_languages[LANGUAGE_TCB])
+		vr_mob = null
+		to_chat(src, span("notice", "System exited safely, we hope you enjoyed your stay."))
+	if(old_mob)
+		old_mob.ckey = ckey
+		ckey = null
+		old_mob.vr_mob = null
+		speech_synthesizer_langs = list(all_languages[LANGUAGE_TCB])
```

Comment:
You do this whole ckey = ckey ckey = null thing a lot, this could be put on a mob proc as well